posthog/test/test_feature_flags.py

```diff
@@ -0,0 +1,3078 @@
+
+    @mock.patch.object(Client, "capture")
+    @mock.patch("posthog.client.decide")
+    def test_get_all_flags_with_no_fallback(self, patch_decide, patch_capture):
+        patch_decide.return_value = {"featureFlags": {"beta-feature": "variant-1", "beta-feature2": "variant-2"}}
+        client = self.client
+        client.feature_flags = [
+            {
+                "id": 1,
+                "name": "Beta Feature",
+                "key": "beta-feature",
+                "is_simple_flag": False,
+                "rollout_percentage": 100,
+                "filters": {
+                    "groups": [
+                        {
+                            "properties": [],
+                            "rollout_percentage": 100,
+                        }
+                    ]
+                },
+            },
+            {
+                "id": 2,
+                "name": "Beta Feature",
+                "key": "disabled-feature",
+                "is_simple_flag": False,
+                "filters": {
+                    "groups": [
+                        {
+                            "properties": [],
+                            "rollout_percentage": 0,
+                        }
+                    ]
+                },
+            },
+        ]
+        self.assertEqual(client.get_all_flags("distinct_id"), {"beta-feature": True, "disabled-feature": False})
+        # decide not called because this can be evaluated locally
+        self.assertEqual(patch_decide.call_count, 0)
+        self.assertEqual(patch_capture.call_count, 0)
+
+    @mock.patch("posthog.client.Poller")
+    @mock.patch("posthog.client.get")
+    def test_load_feature_flags(self, patch_get, patch_poll):
+        patch_get.return_value = {
+            "flags": [
+                {"id": 1, "name": "Beta Feature", "key": "beta-feature", "active": True},
+                {"id": 2, "name": "Alpha Feature", "key": "alpha-feature", "active": False},
+            ],
+            "group_type_mapping": {"0": "company"},
+        }
+        client = Client(FAKE_TEST_API_KEY, personal_api_key="test")
+        with freeze_time("2020-01-01T12:01:00.0000Z"):
+            client.load_feature_flags()
+        self.assertEqual(len(client.feature_flags), 1)
+        self.assertEqual(client.feature_flags[0]["key"], "beta-feature")
+        self.assertEqual(client.group_type_mapping, {"0": "company"})
+        self.assertEqual(client._last_feature_flag_poll.isoformat(), "2020-01-01T12:01:00+00:00")
+        self.assertEqual(patch_poll.call_count, 1)
+
+    def test_load_feature_flags_wrong_key(self):
+        client = Client(FAKE_TEST_API_KEY, personal_api_key=FAKE_TEST_API_KEY)
+        with freeze_time("2020-01-01T12:01:00.0000Z"):
+            self.assertRaises(APIError, client.load_feature_flags)
+
+    @mock.patch("posthog.client.decide")
+    @mock.patch("posthog.client.get")
+    def test_feature_enabled_simple(self, patch_get, patch_decide):
```

Comment:
this and following tests were copied over from the client, so not necessary to write again in every other lib (these libs should have their own old tests 🤷).